google/cloud/security/scanner/audit/fw_rules_engine.py

```diff
@@ -216,23 +213,24 @@ def add_org_policy(self, org_def):
             ids = resource.get('resource_ids', [])
             rules = resource.get('rules', {})
             groups = rules.get('group_ids', [])
-            expanded_rules = []
-            for group in groups:
-                if group not in self.rule_groups_map:
+            expanded_rules = set()
+            for group_id in groups:
+                if group_id not in self.rule_groups_map:
                     raise GroupDoesntExistError(
-                        'Group "%s" does not exist' % group)
-                expanded_group = self.rule_groups_map.get(group, [])
-                expanded_rules.extend(expanded_group)
+                        'Group "%s" does not exist' % group_id)
+                expanded_group = self.rule_groups_map.get(group_id, [])
+                expanded_rules.update(expanded_group)
             for rule_id in rules.get('rule_ids', []):
-                expanded_rules.append(rule_id)
                 if rule_id not in self.rules_map:
                     raise RuleDoesntExistError(
                         'Rule id "%s" does not exist' % rule_id)
+                expanded_rules.add(rule_id)
             for resource_id in ids:
                 gcp_resource = resource_util.create_resource(
                     resource_id=resource_id,
                     resource_type=resource_type)
-                self.org_policy_rules_map[gcp_resource] = expanded_rules
+                self.org_policy_rules_map[gcp_resource] = sorted(
+                    list(expanded_rules))
```

Comment:
nit: sorted will return a list from a set without first converting it to a list. You can just say sorted(expanded_rules).

Reply:
Done.